source/common/ur_adapters_detector.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+#ifndef UR_ADAPTERS_DETECTOR_HPP
+#define UR_ADAPTERS_DETECTOR_HPP 1
+
+#include <array>
+
+#include "logger/ur_logger.hpp"
+#include "ur_util.hpp"
+
+namespace loader {
+
+class AdaptersDetector {
+  public:
+    std::vector<std::string> discoveredPlatforms;
+
+    void discoverEnabledPlatforms() {
```

Comment:
Why not run this code in the constructor? This function should only be called once anyway - otherwise you'll get duplicated entires.

Reply:
Alternatively, you could make this function static (or freestanding, do you even need a class here?) and just return a vector or platforms - this way, even if you call it twice, you get good result.

Reply:
AdapterRegistry is an object with an iterator-like behavior now and it's placed in the context_t.